src/ui/widgets/Image/image.tsx

```diff
@@ -43,8 +51,11 @@ export const ImageComponent = (
         src={file}
         alt={props.alt || undefined}
         style={{
-          height: imageSize,
-          width: imageSize
+          height: props.height ?? imageSize,
+          width: props.width ?? imageSize,
```

Comment:
Here we have a problem. If the widget width is e.g. `50%`, it will have a container div of that size but this 50% will be 50% of that container so will end up as 25%. Try setting 50% on your test widget.

What were you aiming for with these lines?

Reply:
What do you think the `fill` prop was aiming for? Stretch to fit? We should change it to align with the CS-Studio props.

Reply:
I mean change the name.

Reply:
You are right it does make it 25%, but I don't understand why that is happening at all. My goal was just to use the defined height if it is available otherwise default back to the old behaviour. I haven't changed the old behaviour so why is that happening?

I can change the name for the fill prop I assumed it was aiming to just display the image with it's normal size, that makes my other comment redundant which is good.

Reply:
I'm not sure if I have misunderstood you, but the fill prop does not have the same result as the stretchToFit does in CS-Studio.

Reply:
Try `height: "300px", width: "300px"` in the diamond logo on the widget test page.

Reply:
This is not a simple corner of the code, because of the way CSS positioning works. I think `fill` is intended to mean `stretch_to_fit`, because if you set both height and width of an image widget the widget (before your changes) will stretch it. Even with pngs there are lots of cases to think about, svgs possibly even more.

I don't want to hold up this PR with that, so I'll have a look at the rest of it.

Reply:
I have tried that but I'm not sure what the significance of what I'm seeing is. I see what you mean about fill intending to be `stretch_to_fit`, I have changed that on my local version now so can push it, but I thought the way I had done it meant that the previous behaviour would still be applied if the height and width wasn't provided.

Reply:
I think you haven't changed the behaviour if width and height aren't provided, but in the tricky 25% case the width **is** provided.

Reply:
ohhhh I understand, I am a little confused as to why width properties were added to files such as testWidgets.json when there was no width property in the image component though?

Reply:
Positioning is handed in `Widget` as opposed to the individual widgets to avoid having to repeat the code everywhere. To reiterate though, I'm not sure that we've got the best way of handling these positions in all cases so I don't think we should get stuck on this right now. It can be fixed later.